--- ./elements/GL/SoGLClipPlaneElement.cpp.hold	2006-10-18 14:45:02.000000000 -0400
+++ ./elements/GL/SoGLClipPlaneElement.cpp	2006-10-18 15:04:47.000000000 -0400
@@ -35,7 +35,7 @@
 #include <config.h>
 #endif // HAVE_CONFIG_H
 
-#include <Inventor/system/gl.h>
+#include <Inventor/C/glue/gl.h>
 #include <Inventor/errors/SoDebugError.h>
 
 // *************************************************************************
@@ -115,7 +115,7 @@
   GLint val;
   glGetIntegerv(GL_MAX_CLIP_PLANES, &val);
 
-  assert(glGetError() == GL_NO_ERROR &&
+  assert(cc_glue_glGetError()  == GL_NO_ERROR &&
          "GL error when calling glGetInteger() -- no current GL context?");
 
   return (int)val;
