--- ./elements/GL/SoGLModelMatrixElement.cpp.orig	2006-10-18 14:48:29.000000000 -0400
+++ ./elements/GL/SoGLModelMatrixElement.cpp	2006-10-18 15:05:46.000000000 -0400
@@ -38,7 +38,7 @@
 #include <config.h>
 #endif // HAVE_CONFIG_H
 
-#include <Inventor/system/gl.h>
+#include <Inventor/C/glue/gl.h>
 
 #if COIN_DEBUG
 #include <Inventor/errors/SoDebugError.h>
@@ -95,7 +95,7 @@
   
   if (!this->stackoverflow) {
     glPushMatrix();
-    if (glGetError() == GL_STACK_OVERFLOW) {
+    if (cc_glue_glGetError() == GL_STACK_OVERFLOW) {
       this->stackoverflow = TRUE;
     }
   }
