--- ./shapenodes/soshape_bumprender.cpp.orig	2006-10-18 11:30:47.000000000 -0400
+++ ./shapenodes/soshape_bumprender.cpp	2006-10-18 11:32:21.000000000 -0400
@@ -287,7 +287,7 @@
                               (GLsizei)strlen(diffusebumpdirlightvpprogram),
                               diffusebumpdirlightvpprogram);
     GLint errorPos;
-    GLenum err = glGetError();
+    GLenum err = cc_glue_glGetError();
 
     if (err != GL_NO_ERROR) {
       glGetIntegerv(GL_PROGRAM_ERROR_POSITION_ARB, &errorPos);
@@ -302,7 +302,7 @@
     cc_glglue_glProgramString(glue, GL_VERTEX_PROGRAM_ARB, GL_PROGRAM_FORMAT_ASCII_ARB,
                               (GLsizei)strlen(normalrenderingvpprogram),
                               normalrenderingvpprogram);
-    err = glGetError();
+    err = cc_glue_glGetError();
 
     if (err != GL_NO_ERROR) {
       glGetIntegerv(GL_PROGRAM_ERROR_POSITION_ARB, &errorPos);
@@ -343,7 +343,7 @@
     // FIXME: Maybe a wrapper for catching fragment program errors
     // should be a part of GLUE... (20031204 handegar)
     GLint errorPos;
-    GLenum err = glGetError();
+    GLenum err = cc_glue_glGetError();
     if (err != GL_NO_ERROR) {
       glGetIntegerv(GL_PROGRAM_ERROR_POSITION_ARB, &errorPos);
       SoDebugError::postWarning("soshape_bumpspecrender::initPrograms",
@@ -357,7 +357,7 @@
     cc_glglue_glProgramString(glue, GL_VERTEX_PROGRAM_ARB, GL_PROGRAM_FORMAT_ASCII_ARB,
                               (GLsizei)strlen(directionallightvpprogram), directionallightvpprogram);
 
-    err = glGetError();
+    err = cc_glue_glGetError();
     if (err != GL_NO_ERROR) {
       glGetIntegerv(GL_PROGRAM_ERROR_POSITION_ARB, &errorPos);
       SoDebugError::postWarning("soshape_bumpspecrender::initPrograms",
@@ -372,7 +372,7 @@
     cc_glglue_glProgramString(glue, GL_VERTEX_PROGRAM_ARB, GL_PROGRAM_FORMAT_ASCII_ARB,
                               (GLsizei)strlen(pointlightvpprogram), pointlightvpprogram);
 
-    err = glGetError();
+    err = cc_glue_glGetError();
     if (err != GL_NO_ERROR) {
       glGetIntegerv(GL_PROGRAM_ERROR_POSITION_ARB, &errorPos);
       SoDebugError::postWarning("soshape_bumpspecrender::initPrograms",
